src/overlays/actors/ovl_En_Gamelupy/z_en_gamelupy.c

```diff
@@ -32,37 +35,161 @@ const ActorInit En_Gamelupy_InitVars = {
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 10, 30, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80AF6CF0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/EnGamelupy_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/EnGamelupy_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6854.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6944.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6958.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6994.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF69A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6A38.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6A78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/func_80AF6B40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/EnGamelupy_Update.s")
+static TexturePtr sRupeeTex[] = {
+    gameplay_keep_Tex_061FC0, gameplay_keep_Tex_061FE0, gameplay_keep_Tex_062000,
+    gameplay_keep_Tex_062040, gameplay_keep_Tex_062020,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gamelupy/EnGamelupy_Draw.s")
+static Color_RGBA8 sPrimColor = { 255, 255, 255, 255 };
+
+static Color_RGBA8 sEnvColor = { 100, 200, 0, 255 };
+
+void EnGamelupy_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnGamelupy* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.03f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 10.0f);
+    this->actor.shape.yOffset = 700.0f;
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.gravity = -0.5f;
+    this->actor.shape.rot.y = Rand_Next();
+    this->unk_19C = 0;
+    this->unk_19E = 0;
+    this->unk_1A0 = 0x7D0;
+    if (this->actor.params == 1) {
+        this->rupeeIndex = 1;
+    } else {
+        this->rupeeIndex = 0;
+    }
+    func_80AF6944(this);
+}
+
+void EnGamelupy_Destroy(Actor* thisx, PlayState* play) {
+    EnGamelupy* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void func_80AF6854(EnGamelupy* this, PlayState* play) {
+    Vec3f sp4C;
+    Vec3f sp40;
+    Vec3f sp30;
+
+    sp4C = this->actor.world.pos;
+    sp30.x = Math_SinS(this->unk_19E) * 3.0f;
+    sp30.y = 5.5f;
+    sp30.z = Math_CosS(this->unk_19E) * 3.0f;
+    sp40.x = -0.05f * sp30.x;
+    sp40.y = -0.4f;
+    sp40.z = -0.05f * sp30.z;
+    EffectSsKirakira_SpawnDispersed(play, &sp4C, &sp30, &sp40, &sPrimColor, &sEnvColor, 3000, 40);
+    this->unk_19E += this->unk_1A0;
+}
+
+void func_80AF6944(EnGamelupy* this) {
+    this->actionFunc = func_80AF6958;
+}
+
+void func_80AF6958(EnGamelupy* this, PlayState* play) {
+    void* actor = func_800B6680(play, ACTOR_EN_GAMELUPY);
```

Comment:
```suggestion
    s16* unk_198 = func_800B6680(play, ACTOR_EN_GAMELUPY);
```
Now that we know that `this->unk_198` is in fact an `s16*`, we can give the proper type. Also `actor` no longer seems like a reasonable name, so `unk_198` seems fine for the meantime.